src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -197,8 +186,48 @@ public bool HasResponseStarted
             get { return _responseStarted; }
         }
 
+        protected FrameRequestHeaders FrameRequestHeaders => _frameHeaders.RequestHeaders;
+
+        public Frame InitalizeHeaders()
+        {
+            _frameHeaders = HttpComponentFactory.CreateHeaders(DateHeaderValueManager);
+            RequestHeaders = _frameHeaders.RequestHeaders;
+            ResponseHeaders = _frameHeaders.ResponseHeaders;
+            return this;
```

Comment:
On second thought, don't worry about this. I changed it while addressing my own PR feedback readying a merge.